addons/microsoft_calendar/tests/test_create_events.py

```diff
@@ -111,6 +111,32 @@ def test_create_simple_event_from_outlook_attendee_calendar_and_organizer_does_n
         self.assertEqual(len(new_records), 1)
         self.assert_odoo_event(new_records, expected_event)
 
+    @patch.object(MicrosoftCalendarService, 'get_events')
+    def test_create_simple_event_from_outlook_attendee_calendar_where_email_addresses_are_capitalized(self, mock_get_events):
+        """
+        An event has been created in Outlook and synced in the Odoo attendee calendar.
+        The email addresses of the attendee and the organizer are in different case than in Odoo.
+        """
+
+        # arrange
+        outlook_event = dict(self.simple_event_from_outlook_attendee, organizer={
+            'emailAddress': {'address': "Mike@organizer.com", 'name': "Mike Organizer"},
+        }, attendees=[{'type': 'required', 'status': {'response': 'none', 'time': '0001-01-01T00:00:00Z'},
+                       'emailAddress': {'name': 'John Attendee', 'address': 'John@attendee.com'}}])
+
+        mock_get_events.return_value = (MicrosoftEvent([outlook_event]), None)
+        existing_records = self.env["calendar.event"].search([])
+
+        # act
+        self.organizer_user.with_user(self.organizer_user).sudo()._sync_microsoft_calendar()
+
+        # assert
+        records = self.env["calendar.event"].search([])
+        new_records = (records - existing_records)
+        self.assertEqual(len(new_records), 1)
+        self.assert_odoo_event(new_records, self.expected_odoo_event_from_outlook)
+        self.assertEqual(new_records.user_id, self.organizer_user)
+
```

Comment:
Shouldn't we use the microsoft_id field to retrieve the new event coming from microsoft instead of searching all events twice ? Normally these tests are run against database with a low amount of records but sometimes it can be useful to run them against copy of production database. In that case, searching for all events will be a real issue.

Reply:
I figured that as there are already 14 occurrences of full searches in other tests in the same class, performance hasn't really been the focus here. I can optimize this test if you feel it's necessary, but there will still be 10 full table searches for `calendar.event` and 4 full table searches for `calendar.recurrence`.

Reply:
Indeed, you are right. We should create a task to clean that internally at some point.